dunamai/__init__.py

```diff
@@ -479,10 +479,16 @@ def normalize_tag_ref(ref: str) -> str:
     def from_git_tag_topo_order(
         tag_branch: str, git_version: List[int], path: Optional[Path]
     ) -> Mapping[str, int]:
-        cmd = (
-            "git log --simplify-by-decoration --topo-order --decorate=full"
-            ' {} "--format=%H%d"'.format(tag_branch)
-        )
+        if git_version < [2, 10]:
+            cmd = (
+                "git log --simplify-by-decoration --topo-order"
+                ' --decorate=full {} "--format=%H%d"'.format(tag_branch)
```

Comment:
There's certainly room to improve this and avoid duplication, but I've seen this pattern [used below](https://github.com/mtkennerly/dunamai/pull/75/files#diff-30c21a6ebd4b235a4212620f107f6caf6e3de687aa027c3fd64694f88a0a3d86R1118), so I figured I'd stick to it.

Maybe extracting git log commands into a function would help.

Reply:
A way more intrusive change would be to use something like https://github.com/jelmer/dulwich or https://github.com/gitpython-developers/GitPython to avoid having to deal with such details altogether.

Reply:
We can definitely try to reduce some of the duplication, but I think it's okay for now.

I'd like to keep Dunamai's dependencies pretty minimal; otherwise, yeah, Dulwich or GitPython could be nice.